prusti-specs/src/specifications/preparser.rs

```diff
@@ -289,557 +58,547 @@ impl Parse for SpecEntArgs {
+            }
         }
+        self.last_span = span;
+        true
     }
-    /// Convert parsed Rust expression into a Prusti conjunct.
-    fn convert_expr_into_conjunct(&mut self) -> syn::Result<()> {
-        let expr = self.extract_rust_expression()?;
-        self.conjuncts.push(AssertionWithoutId{
-            kind: box common::AssertionKind::Expr(expr)
-        });
-        Ok(())
-    }
-    fn error_expected_expr_without_implication(&self) -> syn::Error {
-        syn::Error::new(self.input.span,
-                        "`==>` cannot be part of Rust expression")
+    /// consume the keyword if it is next in the stream
+    fn consume_keyword(&mut self, keyword: &str) -> bool {
+        if !self.peek_keyword(keyword) {
+            return false;
+        }
+        self.last_span = Some(self.tokens.pop_front().unwrap().span());
+        true
     }
-    fn error_expected_assertion(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected Prusti assertion")
+    /// consume the group if it is next in the stream
+    /// produced its TokenStream, if it has one
+    fn consume_group(&mut self, delimiter: Delimiter) -> Option<TokenStream> {
+        if !self.peek_group(delimiter) {
+            return None;
+        }
+        let token = self.tokens.pop_front().unwrap();
+        let span = token.span();
+        if let TokenTree::Group(group) = token {
+            self.last_span = Some(span);
+            Some(group.stream())
+        } else {
+            None
+        }
     }
-    fn error_expected_operator(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `&&`, `==>`, or `|=`")
+    /// pop a token - note that punctuation is one token per character
+    fn pop(&mut self) -> Option<TokenTree> {
+        if let Some(token) = self.tokens.pop_front() {
+            self.last_span = Some(token.span());
+            Some(token)
+        } else {
+            None
+        }
     }
-    fn error_expected_parenthesis(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `(`")
+    /// pop tokens into a new stream for syn2 until the given operator character
+    fn create_stream_until(&mut self, delimiter: &str) -> TokenStream {
+        let mut t = vec![];
+        while !self.peek_operator(delimiter) && !self.tokens.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        let mut stream = TokenStream::new();
+        stream.extend(t.into_iter());
+        stream
     }
-    fn error_expected_bracket(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `[`")
+    /// pop tokens into a new stream for syn2 until the given operator character
+    fn create_stream_remaining(&mut self) -> TokenStream {
+        let mut t = vec![];
+        while !self.tokens.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        let mut stream = TokenStream::new();
+        stream.extend(t.into_iter());
+        stream
```

Comment:
```suggestion
        TokenStream::from(t.into_iter())
```

Reply:
See above - TokenStream doesn't implement the right `From` trait.